src/components/application_manager/rpc_plugins/app_service_rpc_plugin/src/app_service_app_extension.cc

```diff
@@ -112,6 +116,18 @@ void AppServiceAppExtension::ProcessResumption(
     for (size_t i = 0; i < subscriptions_app_services.length(); ++i) {
       std::string service_type = subscriptions_app_services[i].asString();
       SubscribeToAppService(service_type);
+
+      auto gasd_subscribe = app_mngr::MessageHelper::CreateMessageForHMI(
+          hmi_apis::messageType::request,
+          app_manager_->GetNextHMICorrelationID());
+      (*gasd_subscribe)[app_mngr::strings::params]
+                       [app_mngr::strings::function_id] =
+                           hmi_apis::FunctionID::AppService_GetAppServiceData;
+      (*gasd_subscribe)[app_mngr::strings::msg_params]
+                       [app_mngr::strings::subscribe] = true;
+      (*gasd_subscribe)[app_mngr::strings::msg_params]
+                       [app_mngr::strings::service_type] = service_type;
```

Comment:
@iCollin I believe you could move out all this block of code into another message helper function like `CreateGetAppServiceDataMessage` and pass `bool` value of subscription flag. Then you will have just 2 lines here: create message and manage command

Reply:
I considered making a full helper function for this but the reason I did not is that the function `CreateGetAppServiceDataMessage` would take in both the subscribe bool and the service_type string, meaning it only really helps with the function ID and the keys. But I think you are right because this function could be used in another place (when only app service data subscribed app disconnects we should send GASD(subscribe=false)). I will implement the MessageHelper function soon.

Reply:
Currently I am only budgeted time to review 0190 contribution as this our last sprint of development.

I am fixing the bug simply for now, perhaps I can add this later but there isn't a huge advantage as this is the only place of such a call at the moment and it would need to take in both service_type and subscribe params, so it would just helps with the function ID and keys which we already have included anyways.

Reply:
implemented `MessageHelper::SendGetAppServiceData` in https://github.com/LuxoftSDL/sdl_core/pull/127/commits/3f685192a115b83b1935310965df604dd0d6d3c8